### DISCUSSION TITLE
Port cannot be set for Playwright

### DISCUSSION THREAD

**Discussion by joeinnes** [sveltejs/kit#4806] [Q&A] [answered]
### Describe the bug

I don't seem to be able to change the port that Playwright runs on.

Modifying this value doesn't seem to help: 
https://github.com/sveltejs/kit/blob/143d17f70c525c0a0a929a4e466ada9360523b6a/packages/create-svelte/shared/%2Bplaywright-typescript/playwright.config.js#L5

I also tried updating the previous line to `npm run build && npm run preview --port 3005`, and `PORT=3005 npm run build && npm run preview`, with no joy.

This is frustrating because I can't run `npm run test` while my dev server is running.

### Reproduction

Install SvelteKit with Playwright.
Run `npm run dev`
Run `npm run test`

Watch it fail with an error:
`[WebServer] Port 3000 is occupied`

### System Info

```shell
System:
    OS: Linux 5.16 Pop!_OS 21.10
    CPU: (4) x64 Intel(R) Core(TM) i5-4278U CPU @ 2.60GHz
    Memory: 2.02 GB / 7.68 GB
    Container: Yes
    Shell: 5.1.8 - /bin/bash
  Binaries:
    Node: 17.4.0 - ~/.n/bin/node
    npm: 8.5.1 - ~/.n/bin/npm
  Browsers:
    Brave Browser: beta
    Firefox: 99.0
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.40 
    @sveltejs/kit: next => 1.0.0-next.324 
    svelte: ^3.44.0 => 3.47.0
```

### Severity

annoyance

**Comment by bluwy** [comment, marked as answer]
You need another `--` for `npm run preview`:

```
npm run build && npm run preview -- --port 3005
```

then change to `port: 3005`.

If you want to use the dev server for testing, you can set `webServer.reuseExistingServer: true` too. https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests

**Reply by NatoBoram** [reply]
If anyone is confused about this answer, it's in `playwright.config.ts`:

```typescript
import type { PlaywrightTestConfig } from '@playwright/test';

const config: PlaywrightTestConfig = {
	webServer: {
		command: 'pnpm run build && pnpm run preview --port 3002',
		port: 3002
	}
};

export default config;
```

Both `pnpm run preview --port 3002` and `port: 3002` need to be changed so they can see each other.

(Only `npm` needs an additional `--` to run options like `--port`.)

* https://github.com/sveltejs/kit/issues/5097
* https://github.com/sveltejs/kit/pull/5005
